src/sentry/integrations/jira/notify_action.py

```diff
@@ -4,37 +4,129 @@
 
 from django import forms
 
-from sentry.rules.actions.base import TicketEventAction
 from sentry.models import ExternalIssue
+from sentry.models.integration import Integration
+from sentry.rules.actions.base import TicketEventAction
+from sentry.shared_integrations.exceptions import IntegrationError
 from sentry.utils.http import absolute_uri
 
 logger = logging.getLogger("sentry.rules")
 
 
 class JiraNotifyServiceForm(forms.Form):
-    # TODO 1.0 Add form fields.
+    jira_integration = forms.ChoiceField(choices=(), widget=forms.Select())
 
     def __init__(self, *args, **kwargs):
+        integrations = [(i.id, i.name) for i in kwargs.pop("integrations")]
         super(JiraNotifyServiceForm, self).__init__(*args, **kwargs)
 
-    def clean(self):
-        return super(JiraNotifyServiceForm, self).clean()
+        if integrations:
+            self.fields["jira_integration"].initial = integrations[0][0]
+
+        self.fields["jira_integration"].choices = integrations
+        self.fields["jira_integration"].widget.choices = self.fields["jira_integration"].choices
 
 
 class JiraCreateTicketAction(TicketEventAction):
     form_cls = JiraNotifyServiceForm
-    label = u"TODO Create a {name} Jira ticket"
+    label = u"""Create a Jira ticket in the {jira_integration} account
+    and {project} project
+    of type {issuetype}
+    with components {components}
+    and due date {duedate}
+    with fixVersions {fixVersions}
+    assigned to {assignee}
+    reported by {reporter}
+    label {labels}
+    priority {priority}"""
```

Comment:
Depending on the JIRA instance, I think these required fields could be different. Do we have a plan to handle that?

Reply:
I have a similar concern

Reply:
I was able to refactor the label to only render existing field, but this label will go away when we have the modal.